app/js/components/orderEntry/OrderEntryPage.jsx

```diff
@@ -48,11 +57,92 @@ export class OrderEntryPage extends PureComponent {
     </p>
   );
 
-  switchOrderType = (newOrderType) => {
-    if (!newOrderType) {
-      return this.props.setSelectedOrder({ currentOrderType: {} });
-    }
-    return this.props.setSelectedOrder({ currentOrderType: newOrderType });
+  handleDrugOrderData = (order) => {
+    const {
+      drugDispensingUnits,
+      drugDosingUnits,
+      drugRoutes,
+      durationUnits,
+      orderFrequencies,
+    } = this.props.configurations;
+    const {
+      action,
+      reason,
+      careSetting,
+      dose,
+      dosingUnit,
+      drugInstructions,
+      dosingType,
+      drug,
+      duration,
+      durationUnit,
+      frequency,
+      orderer,
+      dispensingQuantity,
+      dispensingUnit,
+      route,
+      type,
+      previousOrder,
+    } = order;
+
+    return {
+      action,
+      asNeeded: (reason && true) || false,
+      asNeededCondition: reason,
+      autoExpireDate: null,
+      orderReasonNonCoded: null,
+      careSetting,
+      commentToFulfiller: '',
+      dose: dose || null,
+      doseUnits: (dosingUnit && this.getUUID(drugDosingUnits, dosingUnit).uuid) || null,
+      dosingInstructions: drugInstructions || null,
+      dosingType,
+      drug,
+      duration: duration || null,
+      durationUnits: (durationUnit && this.getUUID(durationUnits, durationUnit).uuid) || null,
+      frequency: (frequency && this.getUUID(orderFrequencies, frequency).uuid) || null,
+      numRefills: 0,
+      orderer,
+      previousOrder,
+      quantity: dispensingQuantity || null,
+      quantityUnits:
+        (dispensingUnit && this.getUUID(drugDispensingUnits, dispensingUnit).uuid) || null,
+      route: (route && this.getUUID(drugRoutes, route).uuid) || null,
+      type,
+    };
+  };
+
+  handleLabOrderData = order => ({
+    concept: order.uuid || order.drug,
+    careSetting: this.props.inpatientCareSetting.uuid,
+    encounter: this.props.encounterType.uuid,
+    orderer: this.props.session.currentProvider.uuid,
+    patient: this.props.patient.uuid,
+    type: 'testorder',
+    urgency: order.urgency || 'ROUTINE',
+  });
+
+  handleSubmit = () => {
+    const { draftLabOrders, draftDrugOrders } = this.props;
+    const allDraftOrders = [...draftDrugOrders, ...draftLabOrders.orders];
+    const orders = allDraftOrders.map(order =>
+      (order.type === 'drugorder'
+        ? this.handleDrugOrderData(order)
+        : this.handleLabOrderData(order)));
+
+    const encounterPayload = {
+      encounterProviders: [
+        {
+          encounterRole: this.props.encounterRole.uuid,
+          provider: this.props.session.currentProvider.uuid,
+        },
+      ],
+      encounterType: this.props.encounterType.uuid,
+      location: this.props.session.currentLocation,
+      orders,
+      patient: this.props.patient.uuid,
+    };
+    this.props.createLabOrder(encounterPayload);
```

Comment:
You'll notice that I did not touch the createLabOrder file in this implementation, I only imported it for use, therefore, It would not be ideal to rename it now. there are currently two `actionCreators` to create orders, I decided to just use the lab order implementation, all this would be addressed in the clean up ticket.